### DISCUSSION TITLE
Album in library, but not moved to muisc library directory

### DISCUSSION THREAD

**Discussion by Arskin-Fertrubble** [beetbox/beets#4141] [Q&A] [answered]
My beets configuration for IMPORT is to move files from the source directory to the library directory.

I've just noticed that in some cases beet import reports the album to be already in the library, but it is not moved  from the source to the destination.

Is there a way to force beets to ensure that files to be imported are moved?

NB: I'm confident this is not a permissions issue. 

UPDATE (a few hours later): I notice that when importing many albums that there's a long delay between when the process stage is displayed in the shell and when the files are actually moved to the library folder. I make this as much as  10 minutes or more. If the import is interrupted, i.e. by CTRL+C, the import data processing is way ahead of the files remaining to be moved. It seems those files miss out and stay at the source. Next time I run IMPORT on the same source beets reports the album is already in the library, although it has not been moved to the destination. 
Am I right in the observation? If so, should I  rename & move the  missing album folders manually?  

Thanks

**Comment by Arskin-Fertrubble** [comment, marked as answer]
Worked it out. Don't press CTRL + C multiple times. Just once and allow beets to complete to the point of interrupt.
Although a power failure or PC crash would probably result as described in the OP.

**Comment by wisp3rwind** [comment]
FWIW, below is a reply I had started typing two days ago, but didn't submit yet:

---

> My beets configuration for IMPORT is to move files from the source directory to the library directory.
> 
> I've just noticed that in some cases beet import reports the album to be already in the library, but it is not moved from the source to the destination.

Please, for such questions, provide more details on what you did: Post you configuration (possibly with redacted personal data), post the output of the relevant commands. Without this information, we can only speculate what is going on (but see below: The fact that you Ctrl-C'd the process was crucial missing information).

> Is there a way to force beets to ensure that files to be imported are moved?

That should always happen (unless, say, you specifically skip an album that was detected as duplicate that already is in the library).

> UPDATE (a few hours later): I notice that when importing many albums that there's a long delay between when the process stage is displayed in the shell and when the files are actually moved to the library folder. I make this as much as 10 minutes or more. If the import is interrupted, i.e. by CTRL+C, the import data processing is way ahead of the files remaining to be moved. It seems those files miss out and stay at the source. Next time I run IMPORT on the same source beets reports the album is already in the library, although it has not been moved to the destination. Am I right in the observation? If so, should I rename & move the missing album folders manually?

Yeah, beets processes multiple tasks in parallel: After you select an action for a given album (which is the point when it is displayed on the shell), all remaining processing happens in the background (fetching artwork, lyrics, etc, which might take a while). Files are only moved at the very end of this process. Unfortunately, beets behaviour is not ideal on interrupt wiht Ctrl-C: It essentially stops immediately, possibly leaving half-imported items over (which might already be tracked in the database, but with their files still in their original locations).

To fix this, `beet move`, `beet fetchart`, `beet lyrics` might be helpful to fixup incomplete imports.
